packages/snjs/lib/services/features_service.ts

```diff
@@ -73,9 +77,20 @@ export class SNFeaturesService extends PureService<void> {
       ContentType.ExtensionRepo,
       async (changed, inserted) => {
         const items = [...changed, ...inserted];
-        await this.updateExtensionKeySetting(items);
+        if (this.sessionManager.getUser()) {
+          await this.updateExtensionKeySetting(items);
+        }
       }
     );
+
+    this.removeFullSyncCompletedObserver = this.syncService.addEventObserver(
```

Comment:
@mobitar had to observe the FullSyncCompleted event since LocalDataLoaded was firing after the application was launched, but before sign in. Is this okay?

Reply:
I don't think this is the way to go, especially because you'd be running this logic on every single sync, when it should really be run once in a blue moon.

When you sign in after application launches, the `his.itemManager.addObserver` covers this use case and you'll get notified.

Reply:
That is, after sign in, all ExtensionRepo items will appear in the `inserted` part of that observer.

Reply:
Ahhh so we'd just need the itemManager observer? That's much more simple than I thought, I guess all I needed to do then was remove the migration

Reply:
I think actually we may not need reading local db state. I forget if `inserted` includes items on first load, but you should be able to double check that. If it doesn't, then yes all you need is the observer. If `inserted` doesn't include items on first load, then you would need the observer + reading initial state.

Reply:
I added a test to try this, and `inserted` does seem to include items on first load, but at this point the user isn't yet logged in and the user setting can't be updated. `changed` does fire after login, but more than once, which would mean updating the setting more than once -- not ideal. Not sure how to solve this, since I can't use the LocalDataLoaded event either because it also fires before sign in

Reply:
You’ll want to delete the Repo item after it’s been migrated. Also read the source argument of the observer callback and then only act on a particular source you’re interested in to prevent duplicate calls.

Reply:
New PR here #431, added the item deletion but not sure if the filtered sources are the correct ones